### PR TITLE
[PORT] Silicon Station Alert TGUI and minor fixes!

### PR DESCRIPTION
**Pull request by Bobbanz1**
## About The Pull Request
This PR unifies the station alert menus into one datum system alongside converting the old HTML alert menus into a TGUI version, which removes the HTML alert menus from both the Cyborg and the AI. 

It also fixes two bugs regarding station alerts which apparently kept the station alert console and the alarm modular comp program from having their respective icons updated back to normal once an alert was cleared or added.

The PR also adds two missing SIGNAL_HANDLER thingies to the alarm.dm file.

This PR accomplishes all this by porting the following PRs from TG-Station:
* https://github.com/tgstation/tgstation/pull/61070
* https://github.com/tgstation/tgstation/pull/61475
* https://github.com/tgstation/tgstation/pull/62115 (Atomized?...Version as it only ports the signal handlers that were added to the alarm.dm file)

## Why It's Good For The Game
Makes the Alarm Alert Menu UI look better.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

AI Alarm menu

https://github.com/BeeStation/BeeStation-Hornet/assets/59128051/75f89956-5471-4c02-87b6-0115c468891b

Borg Alarm Menu

https://github.com/BeeStation/BeeStation-Hornet/assets/59128051/f6664149-86aa-4e92-a497-d9f1f1c2f891

</details>

## Changelog
:cl:Arkatos, Mothblocks
refactor: Cyborg Station Alerts UI now uses tgui.
refactor: AI Station Alerts UI now uses tgui.
fix: Station alert console will now always show a correct icon when an alert is added or cleared.
fix: Station alert console and Canary application icons will no longer get stuck in an alert mode forever.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -449,8 +404,6 @@
 	..()
 	if(usr != src)
 		return
-	if (href_list["close"])
-		alerts_popup = null
 	if (incapacitated())
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/ai/ai.dm]: join this call into the above

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 1 above)
Comment (Bobbanz1) [reply code/modules/mob/living/silicon/ai/ai.dm]: > join this call into the above

Could you explain what you want me to do here, do you want me to move the incapacitated call to the topic that opens the ui or move it elsewhere?

Comment (itsmeow) [reply code/modules/mob/living/silicon/ai/ai.dm]: user != src || incapacitated()

Comment (Bobbanz1) [reply code/modules/mob/living/silicon/ai/ai.dm]: Ah thank you, I’ll get it done in the morning

**Comment by mystery3525** [comment]
Tip when you want to break something: Press F7 (buildmode) and right click

**Review by itsmeow — APPROVED** [review]

**Pull request merged**